front/lib/resources/labs_transcripts_resource.ts

```diff
@@ -140,6 +140,14 @@ export class LabsTranscriptsConfigurationResource extends BaseResource<LabsTrans
     return this.update({ isActive });
   }
 
+  async setDataSourceId(dataSourceId: string | null) {
+    if (this.dataSourceId === dataSourceId) {
+      return;
+    }
+
+    return this.update({ dataSourceId });
```

Comment:
Not exactly the interface is sId but then you want to set a foreign key on your model

Reply:
> We def want a foreign key and a number here 

I'm a bit lost - you guys told me we want to store an sId, that we currently don't have in db 
So I can't make an FK on a non-unique field (data_source.name) - what would be the point of having the FK to the .id number if that's not what we're trying to store ? 🤔

Reply:
No no we want sId in APIs, sId or xxxType in internal function interfaces and in DB definitely a foreign key

Reply:
Done - still not a big fan of having the same "id" name convention for 2 different things depending if we're in the interfaces or in the db 😅